app/components/shared/forms/Input.ts

```diff
@@ -407,6 +407,15 @@ export function setPropertiesFormData(obsSource: obs.ISource, form: TFormData) {
     }
   });
 
+  /* Don't fetch properties unless we use it. */
+  if (buttons.length !== 0)
+    properties = obsSource.properties;
```

Comment:
I noticed you omit the curly braces a lot for single line if-statements.  We loosely follow Airbnb's style guide which says that curly braces should only be omitted if the entire statement is a single line.

Here are some examples of what is acceptable and what isn't: https://github.com/airbnb/javascript#blocks

And here's a good explanation of why this style can potentially lead to bugs down the road: https://medium.com/@jonathanabrams/single-line-if-statements-2565c62ff492

I believe Prettier should be doing this automatically, but if it isn't, let me know.